components/http_client.rst

```diff
@@ -1001,6 +1001,21 @@ responses dynamically when it's called::
     $client = new MockHttpClient($callback);
     $response = $client->request('...'); // calls $callback to get the response
 
+If you need to test your error handling with responses that have an HTTP status 
+code different than 200, you need to provide a raw HTTP header with the 
+`response_headers` key::
+
+    use Symfony\Component\HttpClient\MockHttpClient;
+    use Symfony\Component\HttpClient\Response\MockResponse;
+
+    $client = new MockHttpClient([
+        new MockResponse('...', ['response_headers' => ['HTTP/1.1 500 Internal Server Error']]),
```

Comment:
there is a better way, which is to define the `http_code` info instead of `response_headers`